### PR TITLE
fix some problems of golangci-lint

### PR DESCRIPTION
**Pull request by gy95**
Signed-off-by: gy95 <1015105054@qq.com>

modify some codes of golangci-lint
we can use `make lint` command to run golangci-lint tools to check codes.
And too much codes to run all linters, so I only choose `gofmt` `goimports` `golint` `govet` linters to check codes, `goimports` put imports beginning with `volcano.sh` (packages of volcano.sh)after 3rd-party packages;

We can enable more linters to check codes to make codes robuster.

### CODE REVIEW

**Comment by gy95** [comment]
how to fix this e2e error? I just adjust the import package sequence, and  don't modify code logic. Could you please give me some advice? @shinytang6 @Thor-wl

**Comment by shinytang6** [comment]
> how to fix this e2e error? I just adjust the import package sequence, and don't modify code logic. Could you please give me some advice? @shinytang6 @Thor-wl

Yes, I think CI failure is not caused by your changes. E2E testing had some problems which troubled us a lot

**Comment by Thor-wl** [comment]
> how to fix this e2e error? I just adjust the import package sequence, and don't modify code logic. Could you please give me some advice? @shinytang6 @Thor-wl

CI may fails at some cases due to CI machine performance. I'll rerun those jobs.

**Comment by gy95** [comment]
> > how to fix this e2e error? I just adjust the import package sequence, and don't modify code logic. Could you please give me some advice? @shinytang6 @Thor-wl
> 
> CI may fails at some cases due to CI machine performance. I'll rerun those jobs.

Ok, and all ci checks have passed. 👍

File: pkg/controllers/job/plugins/factory.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 	"sync"
 
 	"volcano.sh/volcano/pkg/controllers/job/plugins/env"
-	"volcano.sh/volcano/pkg/controllers/job/plugins/interface"
+	pluginsinterface "volcano.sh/volcano/pkg/controllers/job/plugins/interface"
```
Comment (Thor-wl) [inline pkg/controllers/job/plugins/factory.go:23]: If alias, is there any related code modification?

**Comment by hwdef** [comment]
I don't understand what bad consequences would result from running all the tests.

By the way, https://github.com/golang/lint has been archived, maybe we can use `staticcheck` replace it.

File: pkg/controllers/job/plugins/factory.go (hunk 1 above)
Comment (gy95) [reply pkg/controllers/job/plugins/factory.go:23]: the package name is `pluginsinterface` and the last part of import path `volcano.sh/volcano/pkg/controllers/job/plugins/interface` is `interface`, not equal to package name `pluginsinterface`. So we should use alias. If not, goimports will failed with information File is not `goimports`-ed with -local volcano.sh (goimports)

Comment (jasonliu747) [inline pkg/controllers/job/plugins/factory.go:23]: Modifying `pkg/controllers/job/plugins/factory.go#L40` might be a better choice.

Comment (gy95) [reply pkg/controllers/job/plugins/factory.go:23]: no, this is `goimports` specification, 
the package name is `pluginsinterface` and the last part of import path `volcano.sh/volcano/pkg/controllers/job/plugins/interface` is `interface`, the two are not equal. 
On this senario, we should use alias. 
If not, goimports will failed with information: 
File is not `goimports`-ed with -local volcano.sh (goimports)

**Comment by shinytang6** [comment]
> I don't understand what bad consequences would result from running all the tests.
> 
> By the way, https://github.com/golang/lint has been archived, maybe we can use `staticcheck` replace it.

@gy95 what do you think about it? l think it make sense, btw we can also refer to [test-infra](https://github.com/kubernetes/test-infra/blob/master/.golangci.yml)

**Comment by gy95** [comment]
> > I don't understand what bad consequences would result from running all the tests.
> > By the way, https://github.com/golang/lint has been archived, maybe we can use `staticcheck` replace it.
> 
> @gy95 what do you think about it? l think it make sense, btw we can also refer to [test-infra](https://github.com/kubernetes/test-infra/blob/master/.golangci.yml)

Yes, now that `golang/lint` has been archived, we can use `golangci-lint` to replace it, And there are lots of go projects being using `golangci-lint` tool to do golang code checks, And this tools is much more powerful.

File: .golangci.yml [hunk 2]
```diff
@@ -9,80 +32,35 @@ output:
...
-    - deadcode
-    - errcheck
-    - gocritic
+    # linters maintained by golang.org
     - gofmt
-    - goimports 
-    - golint 
-    - gosimple
+    - goimports
+    - golint
```
Comment (hwdef) [inline .golangci.yml:46]: why keep this archived tools?

File: .golangci.yml [hunk 3]
```diff
@@ -9,80 +32,35 @@ output:
...
-    - interfacer
-    - lll
-    - misspell
-    - staticcheck 
-    - structcheck 
-    - typecheck 
-    - unconvert 
-    - unparam 
-    - varcheck 
+    # linters default enabled by golangci-lint .
```
Comment (hwdef) [inline .golangci.yml:48]: If we want to use golangci-lint, why comment out?

**Comment by gy95** [comment]
> > I don't understand what bad consequences would result from running all the tests.
> > By the way, https://github.com/golang/lint has been archived, maybe we can use `staticcheck` replace it.
> 
> Does anyone have an answer to this question? I don't understand the significance of this PR.

just comply with go programming specifications, but some checks may find some fatal problems, like `gosec` check

File: .golangci.yml (hunk 2 above)
Comment (gy95) [reply .golangci.yml:46]: yes, can be removed

File: .golangci.yml (hunk 3 above)
Comment (gy95) [reply .golangci.yml:48]: linters are too much, if enable all, we should modify more codes, so just first enable some linters, we could open another PR to enable other linters

Comment (hwdef) [reply .golangci.yml:48]: ok, I see.

File: .golangci.yml (hunk 2 above)
Comment (gy95) [reply .golangci.yml]: done

**Review by shinytang6 — APPROVED** [review]
Thanks
/lgtm

**Pull request merged**